R/known-sources.R

```diff
@@ -1,9 +1,52 @@
+# This is a reconstructed schema; it may not be completely accurate, but should
+# be serviceably so. For more information on variables, see the sources below
+# available from Fannie Mae. You may need to create an account first here:
+# https://datadynamics.fanniemae.com/data-dynamics/#/reportMenu;category=HP
+#
+# Data dictionary: https://capitalmarkets.fanniemae.com/resources/file/credit-risk/xls/crt-file-layout-and-glossary.xlsx
+# Names from R code here: https://capitalmarkets.fanniemae.com/media/document/zip/FNMA_SF_Loan_Performance_r_Primary.zip
+fannie_mae_schema <- arrow::schema(
+  LOAN_ID = arrow::string(),
+  ACT_PERIOD = arrow::string(),  # date. Monthly reporting period
```

Comment:
❤️ these comments. This is more a personal preference, but I wonder if they might lock _slightly_ better on the line above? Maybe following the line is the best option, but some of them get kind of long...

Reply:
yeah black messed with some of these on the python side; I'll move them all